### PR TITLE
server() support timeout

### PR DESCRIPTION
**Pull request by TheRatG**
Good day. Thanks for you library. 
Added support timeout command execution. In my case, some command running more than 10 second and deploy is failing.

### CODE REVIEW

File: src/Server/Configuration.php [hunk 1]
```diff
@@ -101,18 +101,42 @@ class Configuration
...
+    /**
      * Construct
      *
      * @param string $name
      * @param string $host
-     * @param int    $port
+     * @param int $port
      */
-    public function __construct($name, $host, $port = 22)
+    public function __construct($name, $host, $port = 22, $timeout = 60)
```
Comment (elfet) [inline src/Server/Configuration.php]: Maybe increase default timeout to 360 sec?

File: src/functions.php [hunk 2]
```diff
@@ -57,15 +58,16 @@ function server($name, $host = null, $port = 22)
 
 /**
  * @param string $name
+ * @param int $timeout
  * @return Builder
  */
-function localServer($name)
+function localServer($name, $timeout = 60)
```
Comment (elfet) [inline src/functions.php]: I do not think `localServer` need this option. We do not connecting to localhost.

File: src/functions.php [hunk 3]
```diff
@@ -33,14 +33,15 @@
  * @param string $name
  * @param string|null $host
  * @param int $port
+ * @param int $timeout
  * @return Builder
  */
-function server($name, $host = null, $port = 22)
+function server($name, $host = null, $port = 22, $timeout = 60)
```
Comment (elfet) [inline src/functions.php]: May do not use past this params to constructor? And just use this setter?
```php
server(...)
    ->connectionTimeout(60);
```

**Comment by elfet** [comment]
And also allow to specify `0` as unlimited timeout.

**Comment by TheRatG** [comment]
Thanks, I will fix it.

File: src/functions.php [hunk 4]
```diff
@@ -65,7 +67,7 @@ function localServer($name)
 
     $env = new Environment();
     $server = new Local();
-    $config = new Configuration($name, 'localhost'); // Builder requires server configuration.
+    $config = new Configuration($name, 'localhost', 22); // Builder requires server configuration.
```
Comment (elfet) [inline src/functions.php]: `22`? Magic number?

**Comment by elfet** [comment]
ping @TheRatG

**Comment by oanhnn** [comment]
This pull request is closed by better pull request #415

**Pull request closed without merging**